--- pari-2.5.2/config/arch-osname.orig	2012-08-17 11:15:49.770079197 -0300
+++ pari-2.5.2/config/arch-osname	2012-08-17 11:16:25.037078238 -0300
@@ -51,6 +51,7 @@ if test -n "$myuname"; then
 	    *RT625*)        arch=sparcv8_super;; # HyperSparc
 	    *CY605*)        arch=sparcv8_super;;
 	    	       esac;;
+		i?86)	arch=i386;;
 	    esac;;
   sunos)    case "$3" in 5*) osname=solaris;; esac
 	    case "$5" in
